src/qs8-packw/gen/qs8-packw-x8c4-gemm-goi-avxvnni-prfm.c

```diff
@@ -0,0 +1,412 @@
+
+        vacc0 = _mm256_dpbusd_avx_epi32(vacc0, vone, v04_0);
+        vacc0 = _mm256_dpbusd_avx_epi32(vacc0, vone, v15_0);
+        vacc0 = _mm256_dpbusd_avx_epi32(vacc0, vone, v26_0);
+        vacc0 = _mm256_dpbusd_avx_epi32(vacc0, vone, v37_0);
+        vacc0 = _mm256_dpbusd_avx_epi32(vacc0, vone, v04_1);
+        vacc0 = _mm256_dpbusd_avx_epi32(vacc0, vone, v15_1);
+        vacc0 = _mm256_dpbusd_avx_epi32(vacc0, vone, v26_1);
+        vacc0 = _mm256_dpbusd_avx_epi32(vacc0, vone, v37_1);
+
+        _mm256_storeu_si256((__m256i *)&out[0],  v04_0);
+        _mm256_storeu_si256((__m256i *)&out[32],  v15_0);
+        _mm256_storeu_si256((__m256i *)&out[64],  v26_0);
+        _mm256_storeu_si256((__m256i *)&out[96],  v37_0);
+        _mm256_storeu_si256((__m256i *)&out[128],  v04_1);
+        _mm256_storeu_si256((__m256i *)&out[160],  v15_1);
+        _mm256_storeu_si256((__m256i *)&out[192],  v26_1);
+        _mm256_storeu_si256((__m256i *)&out[224],  v37_1);
+
+        w0 += 32;
+        w1 += 32;
+        w2 += 32;
+        w3 += 32;
+        w4 += 32;
+        w5 += 32;
+        w6 += 32;
+        w7 += 32;
+        out += 256;
+      }
+
+      // KC main loop multiple of 8x4
+      for (; k >= 4; k -= 4) {
+        __m256i v0 = _mm256_set1_epi32((int32_t) unaligned_load_u32(w0));
+        v0 = _mm256_blend_epi32(v0, _mm256_set1_epi32((int32_t) unaligned_load_u32(w1)), 0x02);
+        v0 = _mm256_blend_epi32(v0, _mm256_set1_epi32((int32_t) unaligned_load_u32(w2)), 0x04);
+        v0 = _mm256_blend_epi32(v0, _mm256_set1_epi32((int32_t) unaligned_load_u32(w3)), 0x08);
+        v0 = _mm256_blend_epi32(v0, _mm256_set1_epi32((int32_t) unaligned_load_u32(w4)), 0x10);
+        v0 = _mm256_blend_epi32(v0, _mm256_set1_epi32((int32_t) unaligned_load_u32(w5)), 0x20);
+        v0 = _mm256_blend_epi32(v0, _mm256_set1_epi32((int32_t) unaligned_load_u32(w6)), 0x40);
+        v0 = _mm256_blend_epi32(v0, _mm256_set1_epi32((int32_t) unaligned_load_u32(w7)), 0x80);
+        xnn_prefetch_to_l1((const int8_t*) w0 + 448);
+        xnn_prefetch_to_l1((const int8_t*) w1 + 448);
+        xnn_prefetch_to_l1((const int8_t*) w2 + 448);
+        xnn_prefetch_to_l1((const int8_t*) w3 + 448);
+        xnn_prefetch_to_l1((const int8_t*) w4 + 448);
+        xnn_prefetch_to_l1((const int8_t*) w5 + 448);
+        xnn_prefetch_to_l1((const int8_t*) w6 + 448);
+        xnn_prefetch_to_l1((const int8_t*) w7 + 448);
+
+        vacc0 = _mm256_dpbusd_avx_epi32(vacc0, vone, v0);
+
+        _mm256_storeu_si256((__m256i *)&out[0],  v0);
+
+        w0 += 4;
+        w1 += 4;
+        w2 += 4;
+        w3 += 4;
+        w4 += 4;
+        w5 += 4;
+        w6 += 4;
+        w7 += 4;
+        out += 32;
+      }
+
+      // KC remainder of 1..3
+      if (k != 0) {
+        assert(k >= 1 && k <= 3);
+
+        const __m256i vmask = _mm256_set1_epi32((1u << (k * sizeof(int8_t) * 8)) - 1);
+
+        __m256i v0 = _mm256_set1_epi32((int32_t) safe_load_32bits(w0, k));
```

Comment:
I think this function should follow the naming conventions of sse or the function it replaces
unaligned_load_u32
becomes
safe_load_u32